src/id.rs

```diff
@@ -84,6 +84,12 @@ impl ToString for Id {
     }
 }
 
+impl std::fmt::Debug for Id {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_tuple("Id").field(&self.to_string()).finish()
```

Comment:
`Id("9m4e2mr0ui3e8a215n4g")` looks like a newtype for string. How about this?

```rust
impl fmt::Debug for Id {
    #[inline]
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        write!(f, "xid:{}", self)
    }
}
```

```rust
println!("{:?}", id); // xid:9m4e2mr0ui3e8a215n4g
```

You'll need to rebase this to get `Display`.

Reply:
> `Id("9m4e2mr0ui3e8a215n4g")` looks like a newtype for string. How about this?
> 
> ```rust
> impl fmt::Debug for Id {
>     #[inline]
>     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
>         write!(f, "xid:{}", self)
>     }
> }
> ```
> 
> ```rust
> println!("{:?}", id); // xid:9m4e2mr0ui3e8a215n4g
> ```
> 
> You'll need to rebase this to get `Display`.

My first reaction was to think "is this Rust idiomatic?". In my experience all the tuple structs I've seen with Debug implemented use the format `Name(value)`. To me this makes it apparent what it is, and what value it has.

Having something like `xid:{}` would suggest that if you do `id.to_string()` that's also what you get since the Debug printed value looks like a string after all?

Reply:
> My first reaction was to think "is this Rust idiomatic?"

I think idiomatic Rust is deriving `Debug` or custom `Debug` with similar output, and showing the internals. In this case, we decided not to for consumer's convenience, so we should optimize for that.

> In my experience all the tuple structs I've seen with `Debug` implemented use the format `Name(value)`. To me this makes it apparent what it is, and what value it has.

That's the problem I mentioned above. `Id("9m4e2mr0ui3e8a215n4g")` is what you get for
```rust
#[derive(Debug)]
struct Id(String);
```
which is misleading.

> Having something like `xid:{}` would suggest that if you do `id.to_string()` that's also what you get since the Debug printed value looks like a string after all?

I don't think so? I'm fine without the prefix (just `Display`) if that bothers you.